src/BridgedRoom.ts

```diff
@@ -648,13 +650,21 @@ export class BridgedRoom {
                         msgtype: "m.text",
                     };
                     await ghost.sendMessage(this.matrixRoomId, messageContent, channelId, eventTS);
-                    // TODO: Currently Matrix lacks a way to upload a "captioned image",
-                    //   so we just send a separate `m.image` and `m.text` message
-                    // See https://github.com/matrix-org/matrix-doc/issues/906
-                    if (message.text) {
-                        return ghost.sendText(this.matrixRoomId, message.text, channelId, eventTS);
-                    }
                 } else {
+                    if (uploadSize && file.size > uploadSize) {
+                        const link = file.public_url_shared ? file.permalink_public : file.url_private;
+                        if (link) {
+                            log.info("File too large, sending as a link");
+                            const messageContent = {
+                                body: `${link} (${file.name})`,
+                                format: "org.matrix.custom.html",
+                                formatted_body: `<a href="${link}">${file.name}</a>`,
+                                msgtype: "m.text",
+                            };
+                            await ghost.sendMessage(this.matrixRoomId, messageContent, channelId, eventTS);
+                        }
```

Comment:
Is it ever possible for `link` to not exist? Is there a need for an else condition?